extmod/modbluetooth_nimble.c

```diff
@@ -616,6 +616,20 @@ int mp_bluetooth_gatts_set_buffer(uint16_t value_handle, size_t len, bool append
 
 #if MICROPY_PY_BLUETOOTH_ENABLE_CENTRAL_MODE
 
+STATIC void gattc_on_data_available(uint16_t event, uint16_t conn_handle, uint16_t value_handle, const struct os_mbuf *om) {
+    MICROPY_PY_BLUETOOTH_ENTER
```

Comment:
Unfortunately this ENTER/EXIT pair must be called by the code here, rather than that in `modbluetooth.c`

Reply:
Yeah... not awesome but I see why.

Maybe could you add a comment to modbluetooth.h above these three functions that the lock must be held?

Otherwise this PR looks good. Thanks!

Reply:
Comment added.